### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by amannn**
This package offers a tiny API surface that can be used for consistent logging across all our node.js apps.

It's a small wrapper around the [winston](https://www.npmjs.com/package/winston) package, which seems to be most popular option for node.js logging (6,104,061 downloads in the last month, followed by [bunyan](https://www.npmjs.com/package/bunyan) with 1,908,126 downloads in the last month).

winston supports logging different levels with different transports: e.g. info logs are printed to the console while critical errors could be reported to sentry (which in turn sends an email notification to me). This isn't implemented yet though, currently we only log to `stdout`.

I had to workaround one issue though: winston expects levels to conform to [RFC 5424](https://tools.ietf.org/html/rfc5424), which specifies that lower levels correspond to higher severity (see page 10. As we need the opposite order, there's some mapping in place. This doesn't concern the public API of this package though.

In regards to JSON fields, I omitted the proposed `logger_name` property as this apparently has ["severe performance implications"](https://github.com/winstonjs/winston/issues/200#issuecomment-173475163) in node.js – I hope that's ok.

In exchange I added a `meta` property which can contain additional information based on the type of log entry (e.g. the status code for HTTP requests).

I'll add another PR to one of our apps where the logger can be seen in action.

### CODE REVIEW

**Comment by sfussenegger** [comment]
the `logger_name` isn't really what you linked to (file, function, line number) but basically just an arbitrary value. it's just that it's common in java to use the class name as logger name:

```
public class MyClass {
  private static final Logger log = LoggerFactory.getLogger(MyClass.class);
  // ...
}
```

this makes it easy to get hierarchical logging system using `.` separated package names allowing queries in kibana for `at.molindo.foo` showing log messages from `at.molindo.foo.Bar` and `at.molindo.foo.baz.Qux`. So that's a very useful feature once the number of messages rises. I'd therefore recommend adding this field as an additional constructor attribute (`const logger = new Logger({service: 'pizza-shop', logger: 'access'});`

in case `logger` property isn't set the `logger_name` attribute can remain empty.

**Comment by amannn** [comment]
Ok, I think I understand.

The node logger is intended to be used as a singleton across a whole app, so there shouldn't be multiple instances with different logger names. But I can allow for an additional parameter when logging to achieve this:

```js
logger.info('Pizza can be served now', {logger: 'cook'});
```

If a module logs a few things, it can also use partial application to specify the logger only once.

In JavaScript I don't know of something like that dot separated class hierarchy as it's used in Java, so I'm not sure if we can achieve that particular use case. If we really need it, it would be a convention that modules use as their logger name.

I'm also not sure how much logging I should add to apps. The most frequent non-error log will probably be server request logs. I could at least add a logger name for those, as they're configured in this package.

File: README.md [hunk 1]
```diff
@@ -1 +1,51 @@
...
\ No newline at end of file
+# molindo-node-logger
+
+A node.js logger that integrates well with the Molindo infrastructure.
+
+## Features
+ - Pretty prints messages in development mode.
+ - Prints JSON messages in production mode.
+ - Errors are printed to `stderr`, while all other levels are printed to `stdout`.
+ - In development only warnings and errors are printed, while in production all levels are printed.
```
Comment (amannn) [inline README.md]: This is actually outdated. In development infos, warnings and errors are printed now. I downgraded HTTP requests from info to debug so they don't show up in development logs.

**Comment by sfussenegger** [comment]
`logger.info('Pizza can be served now', {logger: 'cook'});` seems a bit clunky but it's your call. maybe go for something like `logger.name('cook').info('Pizza can be served now')`

regarding hierarchies, it's only a convention. for instance, if you rest client calls to different services, you could simply use `rest.inquiry` and `rest.shop`

**Comment by amannn** [comment]
I like that builder pattern. I'd have to change some things for that to work properly and I'm not sure how much logging we'll add to node apps, so I'd go for an easier to implement API that is only used internally for the time being.

I just looked into the winston documentation, and they're planning on [overloading the log function](https://github.com/winstonjs/winston#creating-your-own-logger) to accept objects like this in the next major version: 

```js
logger.log({
  level: 'info'
  message: 'Hello distributed log files!',
  // ... other properties
});

// In our case maybe something like:
logger.info({
  message: 'Pizza can be served now',
  name: 'cook'
});
```

I think I'll go for something like that for now. If we need this frequently in consuming apps, we can add a more convenient public API, e.g. with a builder pattern – I hope that's ok.

**Review by sfussenegger — APPROVED** [review]

**Pull request merged**